app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryState.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as Story from '../../../models/beta/story';
+
+const ExpandedStoryState = (props) => {
+  const { story, onEdit } = props;
+
+  return (
+    <div className="Story__section">
+      <div className="Story__section-title">
+        { I18n.translate('activerecord.attributes.story.state') }
+      </div>
+
+      <select
+        value={story._editing.state}
+        className="form-control input-sm"
+        onChange={(event) => onEdit({ state: event.target.value })}
+      >
+        {
+          Story.states.map((state) => (
+            <option value={state} key={state}>
+              { I18n.t(`story.state.${state}`) }
+            </option>
+          ))
+        }
+      </select>
+    </div>
+  );
+};
+
+ExpandedStoryState.propTypes = {
+  story: PropTypes.object
```

Comment:
I think `story` should be `.required` here or have default value in `defaultProps`. Also it is good to add `onEdit` here as well. I'd say that ideally `PropTypes.object` is not a good type (https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-prop-types.md) as it is kinda vague, but that's more of a personal thing.

Reply:
Yes, I agree :smile:.

We gonna change this proptypes to match with the required params of the object using shape or something like that in a future refactoring.